src/test/scala/org/broadinstitute/dsde/firecloud/service/ExportEntitiesByTypeServiceSpec.scala

```diff
@@ -121,6 +122,33 @@ class ExportEntitiesByTypeServiceSpec extends BaseServiceSpec with ExportEntitie
       }
     }
 
+    "when calling GET on exporting a non-FC model entity type with namespaced attributes" - {
+      "OK response is returned and file is entity type when model is flexible" in {
+        Get(namespacedEntitiesTSVPath + "?model=flexible") ~> dummyUserIdHeaders("1234") ~> sealRoute(exportEntitiesRoutes) ~> check {
+          handled should be(true)
+          status should be(OK)
+          entity shouldNot be(empty) // Entity is the first line of content as output by StreamingActor
+          chunks shouldNot be(empty) // Chunks has all of the rest of the content, as output by StreamingActor
+          headers.contains(HttpHeaders.Connection("Keep-Alive")) should be(true)
+          headers should contain(HttpHeaders.`Content-Disposition`.apply("attachment", Map("filename" -> "study.tsv")))
+          contentType shouldEqual ContentType(MediaTypes.`text/tab-separated-values`, HttpCharsets.`UTF-8`)
+          validateLineCount(chunks, MockRawlsDAO.namespacedEntities.length)
+
+          // confirm headers
+          entity.asString.trim.split('\t') should contain theSameElementsAs(List("entity:study_id") ++ MockRawlsDAO.namespacedMetadata("study").attributeNames)
+
+          // columns are returned in a deterministic but not easily-predictable order (hashmaps are involved deep inside).
+          // so it's very difficult for this unit test to check ordering of TSV values. If this test starts failing myseteriously,
+          // it is possible a Java- or Scala- internal ordering changed. To be more lenient, use theSameElementsAs instead of
+          // theSameElementsInOrderAs in the following assertions.
+          val bodyLines = asStringBody(chunks)
+          bodyLines.length shouldBe MockRawlsDAO.namespacedEntities.length // effecively the same assertion as validateLineCount
+          bodyLines.head.split("\t") should contain theSameElementsInOrderAs(List("first", "default-foovalue", "namespaced-foovalue", "first-id"))
```

Comment:
per your comment, i'm slightly worried about the magic ordering of this. should we just future-proof it now?

Reply:
@MatthewBemis I rewrote the test so that it doesn't rely on a pre-existing order, but it still validates the columns/values match. Take another look?